app/helpers/pageflow/social_share_helper.rb

```diff
@@ -24,8 +24,17 @@ def social_share_page_title(page)
     end
 
     def social_share_page_description(page)
-      return page.configuration['text'] if page.configuration['text'].present?
-      return page.configuration['description'] if page.configuration['description'].present?
+      return strip(page.configuration['text']) if page.configuration['text'].present?
+      return strip(page.configuration['description']) if page.configuration['description'].present?
+      social_share_entry_description(page.chapter.revision)
+    end
+
+    def social_share_entry_description(entry)
+      return strip(entry.summary) if entry.summary.present?
+
+      entry.pages.each do |page|
+        return strip(page.configuration['text']) if page.configuration['text'].present?
+      end
       ''
```

Comment:
Given the increasingly complex logic of these methods, I think, we should add some tests.